### PR TITLE
Candid UI - transferring from agent-js

### PR DESCRIPTION
**Pull request by krpeacock**
This introduces a new <candid-ui> element that can be dropped into web pages. It extends the existing functionality of Candid UI, and adds new features like:
- Setting an Identity or Agent
- Changing canister ID's
- Submitting forms by hitting enter
- slots for title, description, and styles

For more info, check the Readme. 

Demo to play with is hosted at https://q27bt-laaaa-aaaab-qaiaa-cai.ic0.app/

### CODE REVIEW

**Comment by krpeacock** [comment]
Some next steps that the community has requested:
- support for inputting arrays in a single input
- blob input through encoded strings or file input

**Comment by chenyan-dfinity** [comment]
I usually test the UI with this canister: https://github.com/chenyan2002/candid-ui-demo. It contains all the Candid types.

File: packages/candid-ui/README.md [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+To use the component, you can either install it from npm:
+
+```bash
+npm install @dfinity/candid-ui
+```
+
+However, the package is optimized to be used in a script tag, so we recommend using the auto setup script from unpkg:
+
+```html
+<script type="module" src="https://unpkg.com/@dfinity/candid-ui/dist/auto.js"></script>
```
Comment (nathanosdev) [inline packages/candid-ui/README.md:29]: Using a CDN introduces a centralized component to a FE app. CDNs are entirely outside of our control, they can pull packages (which is extremely unlikely to be fair) or track users. I'm not against providing documentation for how to use it, but I'm not sure that we should be saying "we recommend".

File: packages/candid-ui/README.md [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+
+```bash
+npm start
+```
+
+### Building
+
+```bash
+npm run build
+```
```
Comment (nathanosdev) [inline packages/candid-ui/README.md:345]: Great docs 👌

**Review by nathanosdev — APPROVED** [review]
I haven't reviewed in very deep detail, but overall it looks good to me 👍

File: packages/candid-ui/README.md (hunk 1 above)
Comment (krpeacock) [reply packages/candid-ui/README.md:29]: We could publish it to a canister "CDN" where people could pull it from

Comment (krpeacock) [reply packages/candid-ui/README.md:29]: For now I changed the language. I'll go ahead and merge and publish, so we can get the base release shipped

File: packages/candid-ui/README.md (hunk 2 above)
Comment (krpeacock) [reply packages/candid-ui/README.md:345]: 💪

**Comment by chenyan-dfinity** [comment]
For nat and int type, I got error: "Do not know how to serialize a BigInt"

File: packages/candid-ui/README.md (hunk 1 above)
Comment (nathanosdev) [reply packages/candid-ui/README.md:29]: I think a canister CDN is a great idea!

**Comment by matthewhammer** [comment]
Cool generalization!

I tried to enter some canisters that were listed on the NNS canister list, and (not surprisingly) some give an error about not implementing the `tmp_hack` needed to get the `.did` file from the canister.

As an extra layer of polish, in some later update, you may want to explore the various ways this fails (like that case) and give another additional user-facing error for each one, or even advice to the canister author about how to remedy it?

**Pull request merged**